server-spi-private/src/main/java/org/keycloak/authorization/policy/evaluation/DefaultPolicyEvaluator.java

```diff
@@ -77,11 +77,6 @@ public void evaluate(ResourcePermission permission, EvaluationContext executionC
             if (resource.getType() != null) {
                 evaluatePolicies(() -> policyStore.findByResourceType(resource.getType(), resourceServer.getId()), consumer);
```

Comment:
@pedroigor 

We identified that this block is also responsible for performances issues when doing entitlement or evaluation when many resources have the same type

Reply:
I've added a comment on https://issues.jboss.org/browse/KEYCLOAK-6621. in general, you should not have too many permissions for a single type. At least not that impact performance so badly as you are suggesting.

I think we can also cache this query. Currently it is hitting the database every time.

Reply:
Actually, results from this method are being cached already.

Reply:
OK for this point. We will not use type anymore for our current usage